mycodo/utils/method.py

```diff
@@ -1,115 +1,94 @@
+        :param method_start_time: when this method started. Must accept datetime and strings
+        :return: float value of the setpoint; True if method finished, otherwise False
+        """
+        return None, False
+
+    def should_restart(self, now, method_start_time=None):
+        """
+        If a method has signalled to be finished, this method is asked if the controller should restart
+        the method processing from the beginning.
+        :param now: point in time to calculate the value for
+        :param method_start_time: when this method started. Must accept datetime and strings
+        :return: True if the method wants to be restarted, otherwise False
+        """
+        return False
+
+    def get_plot(self, max_points_x=None):
+        """
+        Called to calculate values to render a plot in the frontend.
+        :param max_points_x:
+        :return:  2d array with x and y values.
+        """
+        return []
+
+
+class DateMethod(AbstractMethod):
     """
-    Calculates the setpoint from a method
-    :param method_id: ID of Method to be used
-    :param table: Table of the this_controller using this function
-    :param this_controller: The this_controller using this function
-    :param Method: Method SQL table
-    :param MethodData: MethodData SQL table
-    :param logger: The logger to use
-    :return: 0 (success) or 1 (error) and a setpoint value
+    A time/date method allows a specific time/date span to dictate the setpoint.
+    This is useful for long-running methods, that may take place over the period of days, weeks, or months.
     """
-    method = db_retrieve_table_daemon(Method)
 
-    method_key = method.filter(Method.unique_id == method_id).first()
+    def ignore_date(self):
+        """
+        :return: False -> date part of date-time should be considered
+        """
+        return False
 
-    method_data = db_retrieve_table_daemon(MethodData)
-    method_data = method_data.filter(MethodData.method_id == method_id)
+    def calculate_setpoint(self, now, method_start_time=None):
+        # Calculate where the current time/date is within the time/date method
 
-    method_data_all = method_data.filter(MethodData.output_id.is_(None)).all()
-    method_data_first = method_data.filter(MethodData.output_id.is_(None)).first()
+        if self.ignore_date():
+            now = datetime.datetime.strptime(str(now.strftime('%H:%M:%S')), '%H:%M:%S')
 
-    now = datetime.datetime.now()
+        for each_method in self.method_data_all:
+            if self.ignore_date():
+                start_time = datetime.datetime.strptime(each_method.time_start, '%H:%M:%S')
+                end_time = datetime.datetime.strptime(each_method.time_end, '%H:%M:%S')
+            else:
+                start_time = datetime.datetime.strptime(each_method.time_start, '%Y-%m-%d %H:%M:%S')
+                end_time = datetime.datetime.strptime(each_method.time_end, '%Y-%m-%d %H:%M:%S')
```

Comment:
"Date Time" and "Daily Date/Time" calculation only differed by how the now-date was interpreted. This is now shared code that only makes a difference in date handling.